### DISCUSSION TITLE
DC. Accumulator bigger than posts amount

### DISCUSSION THREAD

**Discussion by Fybex** [maxi-blocks/maxi-blocks#4808] [Q&A] [answered]
2) is it ok, that if we dont have enogh posts for columns, they are repeating?
https://jam.dev/c/d4eddb96-0e22-4498-bc0b-f97d5264cad0

_Originally posted by @Olekrut in https://github.com/maxi-blocks/maxi-blocks/pull/4701#pullrequestreview-1408967730_

Not 100% sure, but IMO we should prevent the accumulator going higher than the biggest amount of elements, and the Row should do the same: in case we have a Row of 4 and the CL just returns 3 elements, it should become 3. 

_Originally posted by @Naaaaiix in https://github.com/maxi-blocks/maxi-blocks/pull/4701#pullrequestreview-1410912887_

**Comment by Fybex** [comment]
Let's imagine a real example. Some online shop have products and use one column for each product. Also, it uses context loop to increase accumulator with each item. And let's imagine that one row contains 8 columns(products), but for the last row we have only 3 products(3 columns), so currently, all remaining 5 columns will have the same accumulator and content as 3rd column, and I'm not sure that it's a good option. Maybe we can show that content wasn't found, and also add an option to hide this text on the frontend to avoid any confusion for customers (just have empty space).

**Reply by ckp267** [reply, marked as answer]
@Fybex If there is nothing to repeat, show a simple text message in the repeating column. 

"No more [type] to display"  example:  " No more **posts** to display" or "No more **products** to display" 

But only show text in the EE (editing experience). Never in the VE (visitor experience) front end.

This way the creators will know what to do.

**Reply by Fybex** [reply]
> But only show text in the EE (editing experience). Never in the VE (visitor experience) front end.

I agree, but it's not how it works now with similar situations.

EE:
![image](https://user-images.githubusercontent.com/46112160/236809678-7f9ee510-d42c-4a1d-ad7f-d30b857069af.png)

VE:
![image](https://user-images.githubusercontent.com/46112160/236809705-5769e002-b138-4adf-9f99-5f762ebce4bb.png)

Maybe it's different and I'm missing something or we should fix this as well then.

**Comment by Olekrut** [comment]
and need to test case when are not all columns with DC content
Example: row of 5col: 1st, 2nd, 4th, 5th have DC content and 3rd has simple content, how in this case should work repeater...?

**Comment by ckp267** [comment]
Yes must fix this as well. Think about the website owner. Nobody wants to
show text in the VE "no content found". Poor user experience. Bad for the
website owner etc.

**Comment by ckp267** [comment]
From support last week.

   -

   Hi, here a screencast video where I demonstrate what happens with
   dynamic data when I deactivate Maxi Blocks.
   https://www.screenpresso.com/=Yjskf

**Comment by Fybex** [comment]
closing discussion, thanks, created issue from answer https://github.com/maxi-blocks/maxi-blocks/issues/4952
